### PR TITLE
improvement: S3UTILS-79 document followerDiff scan procedure

### PR DESCRIPTION
**Pull request by jonathan-gramain**
In README.md, provide a procedure to run a full scan to detect
metadata discrepancies across active stateful members.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -465,6 +465,217 @@ node followerDiff.js
 
 * **PARALLEL_SCANS**: number of databases to scan in parallel (default 4)
 
+## Scan Procedure
+
+In order to scan a set of raft sessions to detect discrepancies
```
Comment (watsaqat) [inline README.md:470]: iirc, the use-case is to detect leader-follower discrepancies, correct? If so, lets mention that

File: README.md [hunk 2]
```diff
@@ -465,6 +465,217 @@ node followerDiff.js
...
+```
+ssh -n 'root'@'storage-1' 'echo 3 4 5 7 > /tmp/rs-to-scan'
+ssh -n 'root'@'storage-2' 'echo 1 2 3 5 6 7 8 > /tmp/rs-to-scan'
+ssh -n 'root'@'storage-3' 'echo 1 2 3 4 5 6 8 > /tmp/rs-to-scan'
+ssh -n 'root'@'storage-4' 'echo 1 2 3 4 5 6 7 8 > /tmp/rs-to-scan'
+ssh -n 'root'@'storage-5' 'echo 1 2 4 6 7 8 > /tmp/rs-to-scan'
+```
+
+Check that what you get looks like the above, then execute those
+commands on the shell (e.g. by piping with `| bash -x`).
```
Comment (watsaqat) [inline README.md]: maybe adding the full command to be run is better here

File: README.md [hunk 3]
```diff
@@ -465,6 +465,217 @@ node followerDiff.js
...
+  range of keys multiple times outputting to the same digests
+  database, the risk being to render that range of digests much less
+  efficient to optimize the comparisons, or even useless. If an update
+  of the digests database is wanted, it's best to re-create a new one
+  from scratch (e.g. by deleting the old digests directory first).
+
+- The current version of the script does not work with Metadata bucket
+  format v1, only with v0. If the need arises, it could be made to
+  work with v1 with some more work.
+
```
Comment (watsaqat) [inline README.md:736]: a few Qs:
* if any of the steps fail, should the user start over from step 1, 2, etc?
* will the user always know when a step has failed? could failures be silent at any of the steps or will timeouts alert the user?

File: README.md (hunk 1 above)
Comment (jonathan-gramain) [reply README.md:470]: I'd say that the *goal* is to detect any discrepancy between members, and the *means* is checking follower-leader discrepancies, due to how technically this is achievable without introducing downtime.

File: README.md (hunk 2 above)
Comment (jonathan-gramain) [reply README.md]: Agree, I can add the command to run

File: README.md (hunk 3 above)
Comment (jonathan-gramain) [reply README.md:736]: There could be ways to re-do a single step that fails, either by just relaunching it or by doing a first cleanup before re-launching the step. I should give some details about those cases indeed.

I'd need to make sure the provided commands and scripts fail in a clear way (like checking the error messages or exit code).

This also makes me think that even before running the set of steps, there are pre-conditions to be checked, like making sure followers are up-to-date with the leader for example.

File: README.md [hunk 4]
```diff
@@ -465,6 +465,217 @@ node followerDiff.js
...
+```
+
+**If there are more active stateful hosts to scan, continue with Step 4
+on the next active stateful host.**
+
+### Step 8: Gather results
+
+Finally, once all scans have been done on all active stateful hosts,
+gather all diff results for later analysis.
+
```
Comment (tmacro) [inline README.md:699]: Could include a bit about what to expect in the results. Mine are empty files which I assume means that there are no differences. No need to get too technical just something high level about the different cases.

Comment (jonathan-gramain) [reply README.md:699]: I can add a pointer to where the output format is described above in the `followerDiff` tool description. Indeed empty files mean no difference (one difference per line)

File: README.md (hunk 1 above)
Comment (jonathan-gramain) [reply README.md:470]: I'll add an extra paragraph to explain the general idea of the procedure.

**Review by watsaqat — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Pull request merged**